lib/src/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, Input, forwardRef, Provider } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+import { NgClass } from "@angular/common";
+
+export interface RadioGroupItem {
+    /** The label or text to be displayed in the list */
+    label: string;
+    /** optional description to be displayed next to the label */
+    description?: string;
+    /** any value which should be tied to the item */
+    value: any;
+    /** optional disabled flag. Will show the item grayed out and disabled */
+    disabled?: boolean;
+    /** The id or the unique key of the item */
+    key: string;
+}
+
+interface UniqueItem {
+    id: string;
+    optionItem: RadioGroupItem;
+    selected: boolean;
+}
+
+interface DisplayItem extends UniqueItem {
+    className: string;
+}
+
+const CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RadioGroupComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "sebng-radio-group",
+    templateUrl: "./radio-group.component.html",
+    styleUrls: ["./radio-group.component.scss"],
+    providers: [CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR],
+})
+export class RadioGroupComponent implements ControlValueAccessor {
+    // TODO: Add support for custom html as well as string labels?
+    @Input("list")
+    set list(value: RadioGroupItem[]) {
+        this._list = value;
+        this._generateHelperArrays();
+    }
+    get list(): RadioGroupItem[] {
+        return this._list;
+    }
+    private _list: RadioGroupItem[];
+
+    @Input() name?: string;
+    @Input() label?: string;
+    @Input("className")
+    set className(value: string) {
+        if (value) {
+            this._classList = ["radio-group", value];
+        }
+    }
+    private _classList: NgClass["ngClass"] = "radio-group";
+
+    @Input() disabled?: boolean = false;
+    @Input() condensed?: boolean = false;
+    @Input() inline?: boolean = false;
+
+    get classList(): NgClass["ngClass"] {
+        return this._classList;
+    }
```

Comment:
I don't understand why won't you just use `ngClass` in the template, it's more than enough. 👇 
```html
<div *ngIf="list" class="radio-group" [ngClass]="className" [class.disabled]="disabled" [class.condensed]="condensed">
```

Even if you wanted to clean it up using Typescript, you can just do this instead:

```suggestion
    @Input() disabled?: boolean = false;
    @Input() condensed?: boolean = false;
    @Input() inline?: boolean = false;
    @Input() className?: string;

    get classList(): Array<string> {
        return ["radio-group", className];
    }
```

Reply:
You're right this code is overkill. Was doing it to put all the classes in one list to keep it organized but it turned out not the way I hoped :)